src/main/java/com/puppycrawl/tools/checkstyle/utils/CodePointUtil.java

```diff
@@ -63,12 +63,37 @@ public static boolean hasWhitespaceBefore(int index, int... codePoints) {
      */
     public static int[] stripTrailing(int... codePoints) {
         int lastIndex = codePoints.length;
-        while (CommonUtil.isCodePointWhitespace(codePoints, lastIndex - 1)) {
+        while (lastIndex > 0 && CommonUtil.isCodePointWhitespace(codePoints, lastIndex - 1)) {
             lastIndex--;
         }
         return Arrays.copyOfRange(codePoints, 0, lastIndex);
     }
 
+    /**
+     * Removes leading whitespaces.
+     *
+     * @param codePoints array of unicode code points
+     * @return unicode code points array with leading whitespaces removed
+     */
+    public static int[] stripLeading(int... codePoints) {
+        int startIndex = 0;
+        while (startIndex < codePoints.length
+            && CommonUtil.isCodePointWhitespace(codePoints, startIndex)) {
+            startIndex++;
+        }
+        return Arrays.copyOfRange(codePoints, startIndex, codePoints.length);
+    }
+
+    /**
+     * Removes leading and trailing whitespaces.
+     *
+     * @param codePoints array of unicode code points
+     * @return unicode code points array with leading and trailing whitespaces removed
+     */
+    public static int[] trim(int... codePoints) {
+        return stripLeading(stripTrailing(codePoints));
```

Comment:
Should I separate into
```
final int[] strippedCodePoints = stripTrailing(codePoints)
return stripLeading(strippedCodePoints);
```
?

Reply:
This is probably how I would write it (separated) but you are the author of this PR :)